### DISCUSSION TITLE
Need Assistance with Adding 'Vendor Name' to RIC Indication Header

### DISCUSSION THREAD

**Discussion by mohit4444** [srsran/srsRAN_Project#790] [General Help] [answered]
Hello Community,
I’m looking for some guidance on how to add the 'vendor name' to the RIC Indication header. If anyone has experience with this or can point me in the right direction, I would greatly appreciate it.

Thank you in advance for your help!

**Comment by pgawlowicz** [comment, marked as answer]
You can add it here: https://github.com/srsran/srsRAN_Project/blob/main/lib/e2/e2sm/e2sm_kpm/e2sm_kpm_report_service_impl.cpp#L62

**Reply by mohit4444** [reply]
Thank you for your assistance. I’ve implemented the suggested changes, and I can confirm that the vendor name is now being received by the OSC RIC. However, it seems that the OSC RIC is unable to decode it, as shown in the attached images. Could this be an issue with the OSC RIC?

![osc ric](https://github.com/user-attachments/assets/c19c63d8-da16-4dcc-ba96-b0da029dccf3)
![srs ran](https://github.com/user-attachments/assets/7407d4d6-6051-451c-b688-69fbc19358ac)

**Comment by pgawlowicz** [comment]
indeed, it is not decoded, you can add it here:
https://github.com/srsran/oran-sc-ric/blob/main/xApps/python/lib/e2sm_kpm_module.py#L60

**Reply by mohit4444** [reply]
Thank you but I've added the print messages before the xApp callback is called as shown in the image below. I think https://github.com/srsran/oran-sc-ric/blob/1356ee99cec9489acf8906768a7a8c927b07597d/xApps/python/lib/asn1/e2sm_kpm_packer.py#L128 is not able to set the vendorName like it does with colletStartTime.  
![Screenshot from 2024-08-29 17-08-15](https://github.com/user-attachments/assets/c75692b5-5363-461f-9e75-845f7911c361)

**Reply by pgawlowicz** [reply]
Could you check how the RIC indication header looks like in Wireshark?

**Reply by mohit4444** [reply]
Hi @pgawlowicz
Sorry for the late reply. Here's how the RIC indication header looks in Wireshark and how it looks when i print it in the console.
![2](https://github.com/user-attachments/assets/5c53b805-c237-4ad3-b936-8f25c1c1e6c4)
![1](https://github.com/user-attachments/assets/bbe6d955-920d-49aa-94da-9e401798c930)

**Reply by pgawlowicz** [reply]
Did you change the gnb code to include the vendor name string in the RIC indication?

**Reply by mohit4444** [reply]
Yes i have changed the gnb code and the data is passed to the OSC RIC but it is unable to decode only the optional params in the Indication header. The mandatory param colletStartTime is decoded as expected.
![362748696-7407d4d6-6051-451c-b688-69fbc19358ac](https://github.com/user-attachments/assets/2642b8ad-70a5-4acf-b7be-94abe05ea18f)

**Reply by pgawlowicz** [reply]
you need to also change here:

https://github.com/srsran/oran-sc-ric/blob/main/xApps/python/lib/e2sm_kpm_module.py#L60

**Reply by mohit4444** [reply]
I've added the print messages before the xApp callback (extract_hdr_info) is called as shown in the image below. I think https://github.com/srsran/oran-sc-ric/blob/1356ee99cec9489acf8906768a7a8c927b07597d/xApps/python/lib/asn1/e2sm_kpm_packer.py#L128 is not able to set the vendorName like it does with colletStartTime.
![1p](https://github.com/user-attachments/assets/846d9515-a9bb-49dd-be98-29763c3a96d6)

 here's how it prints in the console
![2p](https://github.com/user-attachments/assets/8184f073-9e12-47bc-96ee-fc8d653971fc)

On a side note. I faced a similar issue with the optional param decoding when i used OSC RIC with a third party commercial gNB setup.

The issue was resolved by modifying the following file: https://github.com/srsran/oran-sc-ric/blob/main/xApps/python/lib/asn1/e2sm-kpm-v4.00.asn#L405

We removed the optional parameters so they wouldn’t be considered during the decoding of the RIC Indication header:

**Reply by pgawlowicz** [reply]
hmm, not sure if I understand correctly, you have removed the optional fields from the ans1 of the RIC Indication Header? then probably this is the reason, the vendor name is not decoded.

**Reply by mohit4444** [reply]
You can ignore the side note in the previous reply. The main issue is that the optional params are not being decoded. Is it possible for you to give me an encoded RIC indication header (with vendor name and sender name)? I'll hardcode it and see if it's being decoded. Thank you for your guidance!

**Reply by pgawlowicz** [reply]
It seems the header is not correctly decoded in the ricxappframe module:
https://github.com/o-ran-sc/ric-plt-xapp-frame-py/blob/master/ricxappframe/e2ap/asn1.py#L138

The message received from the socket contains the vendor name string, but after decoding RIC Indication in the ricxappframe module it is missing.

**Reply by mohit4444** [reply]
Exactly! Should we expect a patch release in the near future?

**Reply by pgawlowicz** [reply]
Unfortunately, we do not plan it.

**Reply by mohit4444** [reply]
I understand. Thank you! @pgawlowicz
